lib/spack/spack/package.py

```diff
@@ -1647,8 +1653,31 @@ def log(self):
             # FIXME : this potentially catches too many things...
             pass
 
+        # Archive the whole stdout + stderr for the package
         install(self.log_path, log_install_path)
+        # Archive the environment used for the build
         install(self.env_path, env_install_path)
+        # Finally, archive files that are specific to each package
+        with working_dir(self.stage.source_path):
+            for glob_expr in self.archive_files:
+                files = glob.glob(glob_expr)
+                for file in files:
```

Comment:
typically we try to avoid overriding builtin functions. Not sure how critical this is though...